src/bundle/Controller/User/FocusModeController.php

```diff
@@ -16,22 +16,43 @@
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Matcher\UrlMatcherInterface;
+use Symfony\Component\Routing\RouterInterface;
 
 final class FocusModeController extends Controller
 {
-    private const RETURN_URL_PARAM = 'returnUrl';
+    private const RETURN_PATH_PARAM = 'returnPath';
 
     private UserSettingService $userSettingService;
 
-    public function __construct(UserSettingService $userSettingService)
-    {
+    private UrlMatcherInterface $urlMatcher;
+
+    private RouterInterface $router;
+
+    /** @var iterable<\Ibexa\Contracts\AdminUi\FocusMode\RedirectStrategyInterface> */
+    private iterable $redirectStrategies;
+
+    /**
+     * @param iterable<\Ibexa\Contracts\AdminUi\FocusMode\RedirectStrategyInterface> $redirectStrategies
+     */
+    public function __construct(
+        UserSettingService $userSettingService,
+        UrlMatcherInterface $urlMatcher,
+        RouterInterface $router,
```

Comment:
`RouterInterface` represents both `UrlMatcherInterface` and `UrlGeneratorInterface`.